### ISSUE TITLE
Allow tooltips on <option>

### ISSUE THREAD

**Issue by dwt** [Pylons/deform#378]
On `<select>` controls, there is currently no way to precisely set `title` attributes of `<option>` elements and control precisely what is in them.

I worked around this by using a custom form control that duplicates the template and then customizes it for this bit. Which I think is unoptimal.

(There is a way to get `title`-attributes on `<option>` elements, but it requires the use of the OptGroup class, which is a completely different beast and still doesn't give precise control over the `title` attribute of the option elements alone (and adds the `<optgroup>` elements, which we don't want)

I'm not sure what the best way to do this would be, my workaround was to use a triple of `(label, value, title)` instead of the tuple that is used right now. 

What would you guys be willing to accept? How would you give control to the user for this?

(By the way, I am of course aware that most browser don't render `title` attributes on `<option>` elements, however we are transforming the `<select>` elements with select2 afterwards - also the select2 template you provide also doesn't allow to control `title` attributes.)

**Comment by stevepiercy** [comment]
As you found, browser support for `title` attribute is inconsistent, and there are preferred ways of displaying tooltips.
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#Accessibility_concerns

Can you provide a link to `title` support in the Select2 widget? I can't find it easily:
- https://duckduckgo.com/?q=site%3Ahttps%3A%2F%2Fselect2.org%2F+tooltip
- https://forums.select2.org/search?q=tooltip

Also a couple of screenshots to illustrate what you want would help me visualize the issue.

**Comment by dwt** [comment]
The effect I achieved using the above mentioned workaround is this:
<img width="834" alt="tooltip_on_option" src="https://user-images.githubusercontent.com/57199/46198035-c6abbb00-c30b-11e8-8652-ee99632a67d5.png">

i.e. all the option elements have a `title` attribute that gives additional information about what the chosen element does.

I'm not sure it's even specially documented on the select2 page. What it does is select *all* options on standard `<selects>` and that happens to include to transform them to something the browsers consistently render title attributes (thus negating many of the disadvantages the mozilla article talks about)

Does that help?

Basically, I would like to have a way to have more influence on the built option elements. Bonus points if it enables me to set data-attributes or whatever else I want. Not sure this is viable, but a dictionary that maps `{attr:value}` might make a lot of sense.

**Comment by stevepiercy** [comment]
@dwt I would still accept a PR for your workaround. I've done similar things in a custom template with the tuple for a table of checkboxes. The PR would need to include tests and documentation, and probably a matching functional test in deformdemo.

Alternatively we will be creating a separate repo under Pylons for the community to contribute their custom widgets with or without tests and documentation.

**Comment by dwt** [comment]
I don't have any time for this right now, but if I do get around to work on this again, I'd be happy to provide a pull request.
